### ISSUE TITLE
add parameter to change min default for Remaining and Recovered to displayed

### ISSUE THREAD

**Issue by Shooter3k** [hashcat/hashcat#3589]
**Describe the feature**
Please make the "Remaining" and "Recovered/Time..." lines an optional parameter with the default being the current 1000 hashes so that this can be turned on when a lower hash count is being worked on

**Current behavior**
Currently, when 1000 hashes are in the hash file, the following lines appear: 
Remaining........: 1000 (0%) Digests
Recovered/Time...: CUR:0,N/A,N/A AVG:0,N/A,N/A (Min,Hour,Day)

Currently, when 100 hashes are in the hash file, the lines do not appear because they are below the hard coded 1000 minimum. 

**Expected behavior**
when 100 hashes are in the hash file and the parameter is used and set to a value of 100, the following lines would appear:

Remaining........: 100 (0%) Digests
Recovered/Time...: CUR:0,N/A,N/A AVG:0,N/A,N/A (Min,Hour,Day)

**Comment by Chick3nman** [comment]
This threshold was chosen because under ~1000 it didn't seem necessary to inform the user of those values because you can quickly spot check and do the conversion in your head. For example: 100 hashes w/ 43 cracked? 57 are left. Around the time when I added this several users complained that the status output was too dense/cluttered for simple tasks and unnecessary lines should be reduced so the threshold gating was required. You can alter this value and recompile without issue but I'm not sure I see the case for a dedicated flag to control this threshold.

**Comment by Shooter3k** [comment]
The uses cases are the same as any hash list...... which is why I'm saying the default shouldn't be changed but allowing the person to set it would be ideal. I love the output and understand others may not.

examples: 

If I start a long running task on 999 hashes, go to bed, and come back to a hash list with 58/941. How many cracks per hour is that?

If a session is running and I'm normally getting a few hashes cracked every few minutes and then all the sudden it jumps up and it's cracking more, how much more per minute did it increase?

**Comment by 0xVavaldi** [comment]
I think it's a good chance to make for any count above 1 hash. I can see the use-cases. Though perhaps this shouldn't be an optional parameter, but a default change.

**Comment by PenguinKeeper7** [comment]
> Around the time when I added this several users complained that the status output was too dense/cluttered for simple tasks and unnecessary lines should be reduced

If you want to reduce clutter, remove stuff like `Restore.Sub`, `Candidate.Engine` and maybe the `Hash.Mode` /  `Hash.Target` / `Candidates`, values that very few people will ever use instead of semi-helpful stuff like Recovered/time for > 1 hash

**Comment by Chick3nman** [comment]
> If I start a long running task on 999 hashes, go to bed, and come back to a hash list with 58/941. How many cracks per hour is that?
> If a session is running and I'm normally getting a few hashes cracked every few minutes and then all the sudden it jumps up and it's cracking more, how much more per minute did it increase?

These 2 examples only involve the Recovered/Time portion, which is a separate threshold in the code from Remaining, so we could lower that threshold if it's useful without changing the Remaining value. It's a much older threshold than Remaining, predates it by ~3 years. I could see some use case for that one but even still, 58 cracks / 5 hours = ~11 - 12 cracks per hour seems like easy enough math to spot check. The main 2 reasons I'm hesitant to implement this are the issue of clutter vs usefulness and the overhead of maintaining more flags that potentially affect the output going forward. There's 2 separate thresholds here and with how easy they are to change in the code I'm not sure adding 2 flags makes sense. If it's useful enough to warrant that change then I feel like we should just change the default thresholds.


> If you want to reduce clutter, remove stuff like `Restore.Sub`, `Candidate.Engine` and maybe the `Hash.Mode` / `Hash.Target` / `Candidates`, values that very few people will ever use instead of semi-helpful stuff like Recovered/time for > 1 hash

The difference is that those values must be presented to the user for the information to be present at all within the status. "Remaining hashes" is entirely a convenience field and without it, the user is still given 100% of the information they need to know how many hashes are left to crack, at the expense of the user having to do some simple subtraction of values <1000 in their head.

**Comment by PenguinKeeper7** [comment]
> "Remaining hashes" is entirely a convenience field

As is Session, Hash Mode, Hash Target and Candidate Engine etc. They can just scroll up 10 terminal lines and see what they put in their own command. Maybe Hashcat needs a -v(erbose) flag. Especially for multiple GPUs, so that you can just have total hashrate instead of taking up 2 lines for every GPU. I can see how this would be controversial though so I don't support it fully to be clear

**Comment by ghost** [comment]
I think I have a solution that could satisfy both sides. Let's not clutter the output if stdout is connected to a tty. On the other hand, if stdout is connected to a pipe (named or unnamed) or a pseudo terminal, the person running hashcat is clearly doing some automation (which is what the issue author has in mind, I believe) and could most probably use all the information.
We'll need a way to tell a real terminal from a pseudo terminal, though.

**Comment by PenguinKeeper7** [comment]
If you're using an automated script, `--machine-readable` or `--status-json` would be more appropriate output formats anyway

**Comment by ghost** [comment]
> If you're using an automated script, `--machine-readable` or `--status-json` would be more appropriate output formats anyway

Not necessarily. JSON output doesn't have the `Recovered/Time` values at the moment, for example. Nor they are displayed with `--machine-readable`.

**Comment by Chick3nman** [comment]
> Not necessarily. JSON output doesn't have the `Recovered/Time` values at the moment, for example. Neither they are displayed with `--machine-readable`.

Not really sure if that's relevant to the core issue here but those values are still what I could call "convenience" values. You are given all of the information you would need to recreate those values. The reason they are present in the standard status output is for people to not have to do the math in their heads, e.g. dividing cracks by elapsed time. With automation that's consuming the machine readable or json outputs, you could recalculate those values for essentially 0 cost outside of hashcat.

**Comment by ghost** [comment]
I agree (although `--machine-readable` doesn't seem to print `runtime_start`).
